.github/workflows/tests.yaml

```diff
@@ -101,11 +102,11 @@ jobs:
     steps:
       - name: Check out code from GitHub
         uses: actions/checkout@v5.0.0
-      - name: Set up Python 3.13
+      - name: Set up Python ${{ matrix.python-version }}
         id: python
         uses: actions/setup-python@v6.0.0
         with:
-          python-version: "3.13"
+          python-version: "3.x"
```

Comment:
I don't think we should change this one here. The version was pinned to prevent accidental workflow issues. Using the newest one right aways won't discover any additional user issues.

At worst it would just break our workflow. Better to update it in a dedicated PR.

Reply:
Please read the commit message when reviewing pull requests.
> Automated testing should tell us if things are broken on the latest stable version of Python before users do.

Reply:
You are recommending manual automated over automatic automated testing?!?

Reply:
You keep missing the point...
> It's also not a huge deal if those jobs aren't running the latest Python version.

It is useful information (that we should not sweep under the rug) if those jobs fail on the latest Python version.

Reply:
> You keep missing the point...

Not sure we're talking about the same thing here.

> > It's also not a huge deal if those jobs aren't running the latest Python version.
> 
> It is useful information (that we should not sweep under the rug) if those jobs fail on the latest Python version.
> 
> Those failures will be good discussions in code reviews.

Which code reviews? If the job update updates its Python version, it would just fail if there is an issue with the new one and we would have a broken workflow which doesn't help anyone. Especially as it would only uncover an issue with test dependencies which don't affect users. Yes, it's useful so these can be fixed as well but those will also be discovered when we open a dedicated PR for the Python bump.

Reply:
There is no cron job on this workflow.  That means that it is only run on a human activity (push, pull request, workflow dispatch), which means that failure results should always be reviewed by one or more humans before merging to main.

Let's revert to manual automated testing...